File: demo/src/main/java/exercise1/UserNotFoundExecption.java
package exercise1;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseStatus;

@ResponseStatus(code = HttpStatus.NOT_FOUND)
public class UserNotFoundExecption extends RuntimeException {
	private static final long serialVersionUID = 4532735241282578711L;

	public UserNotFoundExecption() {
	}

	public UserNotFoundExecption(String message) {
		super(message);
	}

	public UserNotFoundExecption(Throwable cause) {
		super(cause);
	}

	public UserNotFoundExecption(String message, Throwable cause) {
		super(message, cause);
	}

}

File: New folder/Demo - Kotlin/src/main/kotlin/demo/UserNotFoundExecption.kt
package demo

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus


@ResponseStatus(code = HttpStatus.NOT_FOUND)
class UserNotFoundExecption : RuntimeException {
    constructor() {}
    constructor(message: String?) : super(message) {}
    constructor(cause: Throwable?) : super(cause) {}
    constructor(message: String?, cause: Throwable?) : super(message, cause) {}

    companion object {
        private const val serialVersionUID = 4532735241282578711L
    }
}
